src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -9182,6 +9182,8 @@ void Unit::Mount(uint32 mount, uint32 VehicleId, uint32 creatureEntry)
     }
 
     RemoveAurasWithInterruptFlags(AURA_INTERRUPT_FLAG_MOUNT);
+    if (GetTypeId() == TYPEID_UNIT)
```

Comment:
Yeah, there's alot of checks like this though, and
```
if (creature->IsAIEnabled)
    creature->AI()->...
```
instead of
```
if (CreatureAI* AI = creature->AI())
    AI->...
```
that makes me unsure of what approach to use :P